src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSession.cs

```diff
@@ -288,5 +297,25 @@ private void EnsureDeltaApplierforSession()
                     ?? _deltaApplierCreator.Value.CreateManagedDeltaApplier(_runtimeVersion);
             }
         }
+
+        public ValueTask<int?> GetTargetLocalProcessIdAsync(CancellationToken cancellationToken)
+        {
+            if (_callback is IProjectHotReloadSessionCallback2 hotReloadState)
+            {
+                return new ValueTask<int?>(hotReloadState.Process?.Id);
+            }
+
+            return new ValueTask<int?>();
+        }
+
+        public ValueTask<string?> GetProjectFullPathAsync(CancellationToken cancellationToken)
+        {
+            if (_callback is IProjectHotReloadSessionCallback2 hrs)
+            {
+                return new ValueTask<string?>(hrs.Project?.FullPath);
+            }
+
+            return new ValueTask<string?>();
+        }
```

Comment:
The variable names here don't match the new interface name, and aren't consistent with other code. The code uses `callback2` above, which seems good, so we could be consistent here.

```suggestion
        public ValueTask<int?> GetTargetLocalProcessIdAsync(CancellationToken cancellationToken)
        {
            if (_callback is IProjectHotReloadSessionCallback2 callback2)
            {
                return new ValueTask<int?>(callback2.Process?.Id);
            }

            return new ValueTask<int?>();
        }

        public ValueTask<string?> GetProjectFullPathAsync(CancellationToken cancellationToken)
        {
            if (_callback is IProjectHotReloadSessionCallback2 callback2)
            {
                return new ValueTask<string?>(callback2.Project?.FullPath);
            }

            return new ValueTask<string?>();
        }
```